File: module_5/python/stopwatch.py
from datetime import datetime
from time import sleep

def stopwatch(time):
    start = datetime.now()
    try: 
        sleep(float(time))
        print('Time difference of ' + str(datetime.now() - start))
    except ValueError:
        print('Your input was not a valid number.')
        stopwatch_input()
    
def stopwatch_input():
    time_input = input("Test waiting time by typing a number in seconds: ")
    stopwatch(time_input)

stopwatch_input()

File: module_5/r/stopwatch.R
stopwatch <- function(){
  
  time <- readline(paste(
    "Test waiting time by typing a number in seconds: ")
  )
  
  print(time)
  
  while(suppressWarnings(is.na(as.numeric(time)))){
    
    time <- readline(paste("That is not a valid input.",
                           "Please input a number instead: ")
    )
    
  }
  
  start <- Sys.time()
  Sys.sleep(time)
  print(Sys.time() - start)

}

stopwatch()